src/utils/hot_restart.jl

```diff
@@ -0,0 +1,32 @@
+function is_hot_restart_available()
+    return !isinteractive() && !isempty(Base.PROGRAM_FILE) 
```

Comment:
@waralex Am I correct that hot reloading is only available when running a Julia app from the command line rather than within the REPL? (Since the session must be non-interactive, and the `PROGRAM_FILE` must have a value in order for `is_hot_restart_available()` to return `true`.)

Reply:
@rpkyle Yes. Because when working from REPL there is no source code that can be monitored for changes